### PR TITLE
Implement processing cats in engine

### PR DESCRIPTION
**Pull request by Arsene-Baitenov**
(implement processing cats in engine)

### CODE REVIEW

**Review by tepa46 — COMMENTED** [review]
Fix commit messages plz

**Review by tepa46 — COMMENTED** [review]
Now it's looks good to me!

File: model/engine/cell.go [hunk 1]
```diff
@@ -20,39 +22,92 @@ func (engine *engine) cell(point geom.Point) int64 {
...
+
+	if !engine.checkCellColumnRow(column, row) {
+		return false, cellDefault
+	}
+	return true, column*engine.numRows + row
 }
 
-func (engine *engine) tryGetUpRighr(cell int64) int64 {
-	return int64(0)
+func (engine *engine) tryGetUpRighrCell(cell int64) (bool, int64) {
```
Comment (tepa46) [inline model/engine/cell.go]: Righr

File: model/engine/processor.go [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+func (engine *engine) proccessPair(cat int64, neighbour int64) {
+	cats := engine.state.Cats()
+	dist := engine.geom.Distance(cats[cat].Point(), cats[neighbour].Point())
+	if dist <= engine.state.RadiusFight() {
+		cats[cat].setStatus(Fighting)
+		cats[neighbour].setStatus(Fighting)
+
+		cats[cat].setFightings(append(cats[cat].Fightings(), neighbour))
+	} else if dist <= engine.state.RadiusHiss() {
+		if engine.rndCore.Float64() >= hissingProbability(dist) {
```
Comment (tepa46) [inline model/engine/processor.go]: < ?

Comment (Arsene-Baitenov) [reply model/engine/processor.go]: 100%

File: model/engine/cell.go (hunk 1 above)
Comment (Arsene-Baitenov) [reply model/engine/cell.go]: Oi

**Review by tepa46 — APPROVED** [review]

**Pull request merged**